pkg/controller/stacks/install/installjob.go

```diff
@@ -328,3 +331,49 @@ func isCRDObject(obj runtime.Object) bool {
 	return apiextensions.SchemeGroupVersion == gvk.GroupVersion() &&
 		strings.EqualFold(gvk.Kind, "CustomResourceDefinition")
 }
+
+func setStackDefinitionControllerEnv(obj *unstructured.Unstructured, namespace, name string) error {
+	env := []corev1.EnvVar{{
+		Name:  stacks.StackDefinitionNamespaceEnv,
+		Value: namespace,
+	}, {
+		Name:  stacks.StackDefinitionNameEnv,
+		Value: name,
+	}}
+
+	// use convert functions because SetUnstructuredContent is unwieldy
+	sd, err := convertToStackDefinition(obj)
+	if err != nil {
+		return err
+	}
+
+	ts := &sd.Spec.Controller.Deployment.Spec.Template.Spec
+	ts.Containers[0].Env = append(ts.Containers[0].Env, env...)
```

Comment:
Containers have `Name` and since we're the ones generating the whole deployment section for template stacks, we know that name. We can filter the containers by `Name` of the one we want and add the env vars to make sure we're adding them to the correct one.